core/domain/exp_services.py

```diff
@@ -1864,24 +1865,25 @@ def is_version_of_draft_valid(exp_id, version):
 def get_user_exploration_data(
         user_id, exploration_id, apply_draft=False, version=None):
     """Returns a description of the given exploration."""
+    exp_user_data = user_models.ExplorationUserDataModel.get(
+        user_id, exploration_id)
     if apply_draft:
-        exploration = get_exp_with_draft_applied(exploration_id, user_id)
+        is_valid_draft_version, exploration = (
```

Comment:
As an FYI, this reads oddly to me. I'd expect get_exp_with_draft_applied to return an exploration and not a 2-tuple.

What if you have get_exp_with_draft_applied() return None or raise exception if the draft can't be applied, and use that instead to infer whether the draft is valid or not?

Reply:
I see, but get_exp_with_draft_applied currently returns a plain exploration if draft can't be applied and I assume we don't want to modify this behavior?

Reply:
I think if you need the correctness signal then it's fine to change as mentioned above, since presumably the caller has access to the original exploration or can call get_exp_by_id(). (At least, that's the cleanest way I can think of to do it.)

Reply:
Thanks, i've adopted your suggestion.